spec/directives/use/load.hrx

```diff
@@ -0,0 +1,453 @@
+a {file: other3}
+
+<===> css_order/use_order/output.css
+a {
+  file: other1;
+}
+
+a {
+  file: other2;
+}
+
+a {
+  file: other3;
+}
+
+a {
+  file: input;
+}
+
+<===>
+================================================================================
+<===> css_order/diamond/input.scss
+@use "left";
+@use "right";
+
+a {file: input}
+
+<===> css_order/diamond/left.scss
+@use "shared";
+
+a {file: left}
+
+<===> css_order/diamond/right.scss
+@use "shared";
+
+a {file: right}
+
+<===> css_order/diamond/shared.scss
+a {file: shared}
+
+<===> css_order/diamond/output.css
+a {
+  file: shared;
+}
+
+a {
+  file: left;
+}
+
+a {
+  file: right;
+}
+
+a {
+  file: input;
+}
+
+<===>
+================================================================================
+<===> css_order/triangle/input.scss
+@use "midstream";
+@use "upstream";
+
+a {file: input}
+
+<===> css_order/triangle/midstream.scss
+@use "upstream";
+
+a {file: midstream}
+
+<===> css_order/triangle/upstream.scss
+a {file: upstream}
+
+<===> css_order/triangle/output.css
+a {
+  file: upstream;
+}
+
+a {
+  file: midstream;
+}
+
+a {
+  file: input;
+}
+
+<===>
+================================================================================
+<===> css_order/unrelated_branches/input.scss
+@use "left_midstream";
+@use "right_midstream";
+
+a {file: input}
+
+<===> css_order/unrelated_branches/left_midstream.scss
+@use "left_upstream";
+
+a {file: left midstream}
+
+<===> css_order/unrelated_branches/left_upstream.scss
+a {file: left upstream}
+
+<===> css_order/unrelated_branches/right_midstream.scss
+@use "right_upstream";
+
+a {file: right midstream}
+
+<===> css_order/unrelated_branches/right_upstream.scss
+a {file: right upstream}
+
+<===> css_order/unrelated_branches/output.css
+a {
+  file: left upstream;
+}
+
+a {
+  file: left midstream;
+}
+
+a {
+  file: right upstream;
+}
+
+a {
+  file: right midstream;
+}
+
+a {
+  file: input;
+}
+
+<===>
+================================================================================
+<===> import_order/use_into_use/input.scss
+@use "midstream";
+
+@import "input.css";
+
+<===> import_order/use_into_use/midstream.scss
+@use "upstream";
+
+@import "midstream.css";
+
+a {file: midstream}
+
+<===> import_order/use_into_use/upstream.scss
+@import "upstream.css";
+
+<===> import_order/use_into_use/output.css
+@import "upstream.css";
+@import "midstream.css";
+@import "input.css";
+
+a {
+  file: midstream;
+}
+
+<===>
+================================================================================
+<===> import_order/import_below_rule/input.scss
+@use "midstream";
+
+@import "input.css";
+
+<===> import_order/import_below_rule/midstream.scss
+@use "upstream";
+
+a {file: midstream}
+
+@import "midstream.css";
+
+<===> import_order/import_below_rule/upstream.scss
+@import "upstream.css";
+
+<===> import_order/import_below_rule/output.css
+@import "upstream.css";
+@import "midstream.css";
+@import "input.css";
+
+a {
+  file: midstream;
+}
+
```

Comment:
Why is  `@import "input.css";` above the `a {file: midstream}` in the output here?

Reply:
CSS `@import`s are required to appear before any other rules (other than `@charset`) in the document. All these `import_order` tests ensure that we maintain that invariant while hewing as closely as possible to the original source order.